### ISSUE TITLE
Style transfer for carton faces

### ISSUE THREAD

**Issue by AstraliteHeart** [weepiess/StyleFlow-Content-Fixed-I2I#2]
Hello and thank you for sharing your work, very impressive results!

I am trying to apply your technique for sim2real style transfer of portraits and would appreciate some pointers on how to improve results.

My original dataset is a collection of synthetic 3d faces 256x256px, 25k samples. 
My target dataset is a collection of digital face paintings also at 256px, 25k samples (with a much larger variety of styles).

I used wikiart config as base and trained my model from scratch for the maximum step amount. I am observing some issues (which perhaps is the same issue in disguise), which also seems to be present if I use your pretrained model.

1. The overall quality of style transfer seems to be lower than I expected based on your presented samples.
2. There are some visual artifacts around borders
3. All generated outputs contain some form of weird dithering/checkerboard artifacts.

Here are some examples:

Wikiart model
[![sample.png](https://i.postimg.cc/4N54Kp6v/sample.png)](https://postimg.cc/svMCqQ01)

My model
[![177800-0-0.jpg](https://i.postimg.cc/02H3HQPR/177800-0-0.jpg)](https://postimg.cc/F73DYhYD)

Any suggestions on what I may be doing wrong?
Thank you.

**Comment by WeichenFan** [comment]
Hi,

Thank you for using our model! 

>  "My target dataset is a collection of digital face paintings also at 256px, 25k samples (with a much larger variety of styles)."

_Would you mind providing a detailed number of styles you are using?_

> "I used wikiart config as base and trained my model from scratch for the maximum step amount. I am observing some issues (which perhaps is the same issue in disguise), which also seems to be present if I use your pretrained model."

_Our model do have some checkerboard phenomenon, especially when the content and target shares very different semantic information.  For visual artifacts or dithering, it is caused by false matching of the style loss. If the pre-trained vgg-19 mismatch the feature when calculating style loss, the generated image would have more artifacts. To solve this problem, we proposed Aligned-Style Loss, which only uses the top-matched feature for style loss calculation._ 

> "I am trying to apply your technique for sim2real style transfer of portraits and would appreciate some pointers on how to improve results."
 
_In your case, it seems that the source and target images share few mutual semantic information, so you could try to lower the keep ratio in the config file. And we have updated the config and Net.py for configurable keep ratio._

---

**- A temporary solution**
Please use the new config file(wikiart_tv_0.6), which replaces the structure-aware loss with tv loss, and lowers the keep ratio. During our experiments, we find that tv loss could ease the checkerboard phenomenon (better visual performance), but the aligned-style loss could achieve higher FID. And we also proved a pre-trained model using wikiart_tv_0.6.yaml.

**- The fundamental solution**
The artifact or checkerboard is actually caused by the squeeze operation (all Flow models have this problem), and we are now working on a model with a new architecture to solve this problem. Code and paper would be released in a short time (would be updated in this repo), feel free to use then :)

link to new ckpt: https://drive.google.com/file/d/15ei5xaNpZ9fnU9YybLK5s8o27guySsMf/view?usp=sharing
